Add support for Ctrl-K

wch opened the pull request:
This PR adds support for Ctrl-K, which deletes from the current cursor position to the end of the line.

wch commented on src/readline.ts:
```diff
@@ -315,6 +315,9 @@ export class Readline implements ITerminalAddon {
         this.state.update("");
         break;
+      case InputType.CtrlK:
+        this.state.editDelete(this.state.buffer().length);
```
Using `this.state.buffer().length` will often be a larger value than necessary, but I did not see a simple way to compute the exact number of characters to delete to the end of the line, and this works correctly in my testing.

Another possibility is to modify the `State` class to do a more precise operation.

strtok replied:
Yeah, this would end up deleting multiple lines in multi line code blocks. I think that adding this to line.ts:

```typescript
  public deleteEndOfLine(): boolean {
    if (this.buf.length == 0 || this.pos == this.buf.length) {
      return false;
    }

    const start = this.pos;
    const end = this.endOfLine();
    if (start == end) {
      this.delete(1);
    } else {
      this.buf = this.buf.slice(0, start) + this.buf.slice(end);
    }

    return true;
  }
  ```
  
  And then adding a supporting function in state.ts called by the ctrl-k code would do it:
  
  ```typescript
  public editDeleteEndOfLine() {
    if (this.line.deleteEndOfLine()) {
      this.refresh();
    }
  }
  ```

If you update the PR I'll merge it! Thanks!

wch commented:
Thanks for the suggestions! I've updated the PR and added some unit tests.

One note: For multi-line input, the up and down cursor movement doesn't account for characters in the prompt string, which results in the cursor moving left or right (in addition to up and down). I wrote the tests to expect this behavior, but if it changes in the future, those tests could break.

wch commented:
Follow-up about the multi-line input: With python, there is a continuation prompt that is indented the same amount as the regular prompt, and maybe that's the way to go (in the future) instead of trying to compensate for differing indentation.

```
>>> def foo():
...   return 1
... 
```
